### DISCUSSION TITLE
RTMP stream not working

### DISCUSSION THREAD

**Discussion by mcarson75** [blakeblackshear/frigate#1049] [Ask A Question]
I have the following roles defined under the input of my cameras:

```
          roles:
            - clips
            - rtmp
            - detect
```

along with 
```
    rtmp:
      # Required: Enable the live stream (default: True)
      enabled: True
```

Under `ffmpeg: output_args:` I have
```
        rtmp: '-c copy -f flv -codec:a aac'
```

However, when I try to view an rtmp stream at `rtmp://<frigate_fqdn>/live/<camera_name>` I have never been able to view anything.  VLC doesn't give any errors, it just doesn't give any output.  I'm not sure if there is something else I need to do?  Is there a username/password for the rtmp stream?

**Comment by paularmstrong** [comment]
Do you have port 1935 open from your Frigate container? https://blakeblackshear.github.io/frigate/configuration/cameras/#rtmp-streams

**Reply by mcarson75** [reply]
Yes I have this in docker-compose:

```
    ports:
      - '5000:5000'
      - '1935:1935' # RTMP feeds
```

and no firewall on the PC.

**Reply by paularmstrong** [reply]
Can you run `ffprobe rtmp://<frigate_fqdn>:1935/live/<camera_name>` and paste the output?

Also, make sure you're including the port `:1935` when you're trying to access through VLC

**Reply by mcarson75** [reply]
Here is the ffprobe output.  I did try VLC both with and without the port specification.

```
ffprobe version 4.2.4-1ubuntu0.1 Copyright (c) 2007-2020 the FFmpeg developers
  built with gcc 9 (Ubuntu 9.3.0-10ubuntu2)
  configuration: --prefix=/usr --extra-version=1ubuntu0.1 --toolchain=hardened --libdir=/usr/lib/x86_64-linux-gnu --incdir=/usr/include/x86_64-linux-gnu --arch=amd64 --enable-gpl --disable-stripping --enable-avresample --disable-filter=resample --enable-avisynth --enable-gnutls --enable-ladspa --enable-libaom --enable-libass --enable-libbluray --enable-libbs2b --enable-libcaca --enable-libcdio --enable-libcodec2 --enable-libflite --enable-libfontconfig --enable-libfreetype --enable-libfribidi --enable-libgme --enable-libgsm --enable-libjack --enable-libmp3lame --enable-libmysofa --enable-libopenjpeg --enable-libopenmpt --enable-libopus --enable-libpulse --enable-librsvg --enable-librubberband --enable-libshine --enable-libsnappy --enable-libsoxr --enable-libspeex --enable-libssh --enable-libtheora --enable-libtwolame --enable-libvidstab --enable-libvorbis --enable-libvpx --enable-libwavpack --enable-libwebp --enable-libx265 --enable-libxml2 --enable-libxvid --enable-libzmq --enable-libzvbi --enable-lv2 --enable-omx --enable-openal --enable-opencl --enable-opengl --enable-sdl2 --enable-libdc1394 --enable-libdrm --enable-libiec61883 --enable-nvenc --enable-chromaprint --enable-frei0r --enable-libx264 --enable-shared
  libavutil      56. 31.100 / 56. 31.100
  libavcodec     58. 54.100 / 58. 54.100
  libavformat    58. 29.100 / 58. 29.100
  libavdevice    58.  8.100 / 58.  8.100
  libavfilter     7. 57.100 /  7. 57.100
  libavresample   4.  0.  0 /  4.  0.  0
  libswscale      5.  5.100 /  5.  5.100
  libswresample   3.  5.100 /  3.  5.100
  libpostproc    55.  5.100 / 55.  5.100
```

EDIT:
Just to give a little more detail - I ran this from three places - from another machine on the same subnet, from the host machine, and from inside the docker container itself (against 127.0.0.1 and the docker address 172.18.0.2).  I got the same output from all.

**Comment by MattEng0** [comment]
I hope it is as simple as this, but have you tried removing the `‘`  from your rtmp output args? Don’t think they are required.

**Reply by mcarson75** [reply]
I tried taking out the `'` as well as commenting out this line completely but I still don't get an rtmp stream and get the same output from ffprobe.

**Comment by Kate222** [comment]
I use user and password with rtmp. I also use some ffmpeg with it. Different camera has different rtmp and ffmepg.

**Reply by mcarson75** [reply]
I am having trouble with RTMP outbound from frigate, not inbound.
